app/models/reduction_fetcher.rb

```diff
@@ -1,49 +1,51 @@
-class ReductionFetcher
-  attr_accessor :topic
-  def initialize(filter)
-    @filter = filter
-    @subject_reductions = SubjectReduction.where(filter.except(:user_id))
-    @user_reductions = UserReduction.where(filter.except(:subject_id))
-
-    @topic = :reduce_by_subject
+class ReductionFetcher < FetcherBase
+  attr_reader :loaded
+
+  def initialize(query)
+    super(query)
+    @loaded = false
+
+    @subject_reductions = SubjectReduction.where(query.except(:user_id))
```

Comment:
the base class handles stuff related to the query because it's done the same for the extract fetcher and reduction fetcher both. it doesn't make sense to hold query in reduction fetcher instead of the base class.